buildconfig/CMake/GNUSetup.cmake

```diff
@@ -115,3 +115,13 @@ if( CMAKE_COMPILER_IS_GNUCXX )
     set( CMAKE_SHARED_LIBRARY_LINK_CXX_FLAGS "${CMAKE_SHARED_LIBRARY_LINK_CXX_FLAGS} -static-libgcc -static-libstdc++" )
   endif()
 endif()
+
+option (COLORED_OUTPUT "Always produce ANSI-colored output (GNU/Clang only)." TRUE)
+
+if (${COLORED_OUTPUT})
```

Comment:
Just a small style point for cmake. You can avoid the `${}` brackets in testing variables as cmake will expand them automatically, for example this is equivalent to `if(COLORED_OUTPUT)`. We tend to use the latter style as it's easier to read.